packages/governance/src/types.js

```diff
@@ -179,6 +252,271 @@
 /**
  * @callback AddQuestion
  * @param {Installation} voteCounter
- * @param {QuestionTermsShort} questionDetailsShort
+ * @param {BallotSpec} ballotSpec
  * @returns {Promise<AddQuestionReturn>}
  */
+
+/**
+ * @typedef QuestionCreator
+ * @property {AddQuestion} addQuestion
+ */
+
+/**
+ * @callback CreateQuestion
+ *
+ * @param {string} name - The name of the parameter to change
+ * @param {ParamValue} proposedValue - the proposed value for the named
+ *   parameter
+ * @param {Installation} ballotCounterInstallation - the ballotCounter to
+ *   instantiate to count votes. Expected to be a binaryBallotCounter. Other
+ *   ballotCounters might be added here, or might require separate governors.
+ *   under management so users can trace it back and see that it would use
+ *   this electionManager to manage parameters
+ * @param {Instance} contractInstance - include the instance of the contract
+ * @param {ClosingRule} closingRule - deadline and timer for closing voting
+ * @returns {Promise<BallotDetails>}
+ */
+
+/**
+ * @typedef {Object} ParamChangeQuestion
+ * @property {ParamSpecification} paramSpec
+ * @property {Instance} contract
+ * @property {ParamValue} proposedValue
```

Comment:
Clearly your intention is that the value of `paramSpec` determines what the correct type of `proposedValue` is. @michaelfig introduced me to a pattern (that I never would have figured out on my own) that I use at https://github.com/Agoric/agoric-sdk/blob/6dc71522df76c5eb1d2decdd3be0d82a98a2727e/packages/marshal/src/types.js#L145 and elsewhere. That one has additional complexity in order to express recursive types, which isn't an issue here. This one should be much simpler. I leave all further explanation to @michaelfig as I still don't really understand how this works.

Reply:
I'm not really sure what to explain here.  I could look at the code to understand further, but not right now.

Reply:
@erights showed how to do this, and that commit has been added to this PR.  Thanks, MarkM!